### PR TITLE
N remover

### PR DESCRIPTION
**Pull request by dstreett**
@joe-angell Another one for you to review once you have a bit. This one will be much nicer than the other one and should go pretty quick! :) 

Thank you, for your time!! :)

### CODE REVIEW

File: N-Remover/src/nremover.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#include <boost/functional/hash.hpp>
+#include <algorithm>
+
+typedef std::unordered_map<std::string, size_t> Counter;
+/*This is a O(N) time algorithm
+ * it will search for the longest base pair segment that has
+ * no N's within it*/
+void trim_n(Read &rb, size_t min_length) {
+
+    std::string seq = rb.get_seq();
```
Comment (joe-angell) [inline N-Remover/src/nremover.h]: You don't need a copy of the seq here, so use `const std::string & seq = rb.get_seq();`

File: N-Remover/src/nremover.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    int len = seq.length() - 1;
+    int bestLeft = 0, bestRight = len + 1;
+
+    for (int i = 0  ; i <= len; ++i) {
+        if (seq[i] == 'N') {
+            if ( (bestLeft - bestRight > bestRight - i || bestLeft - bestRight > bestLeft - i) && (bestLeft != 0 && bestRight != len)  ) {
```
Comment (joe-angell) [inline N-Remover/src/nremover.h]: shouldn't these be `bestRight - bestLeft`?  Recommend abstracting this to a function call like distance, so it's more clear what you are doing.  I would add some more tests for edge cases, like read3 with no N at the end of the read should fail i think.  Also what happens if the read is all 'N'?

maybe rewrite so it's a bit easier to understand, like build the best as you go, not stopping just on 'N', but every character.  So you have a bestLeft, bestRight, currentLeft, currentRight, and update best when `distance(currentLeft, currentRigth) > distance(bestLeft, bestRight)` and start them both at 0.

**Comment by dstreett** [comment]
Hey, @joe-angell, Is this looking better?

**Comment by dstreett** [comment]
Hey, @joe-angell , just a little bump to have you check this out. :) No rush, but wanted to make sure you haven't forgotten.

**Comment by joe-angell** [comment]
Yeah much easier to understand looks good.

**Comment by dstreett** [comment]
Thanks, @joe-angell :)
